crates/catalog/src/mutator.rs

```diff
@@ -74,6 +109,22 @@ impl CatalogMutator {
 
         Ok(state)
     }
+
+    /// Mutate the catalog if possible and immediately commit the changes.
+    ///
+    /// Errors if the metastore client isn't configured.
+    ///
+    /// This will retry mutations if we were working with an out of date
+    /// catalog.
+    pub async fn mutate_and_commit(
+        &self,
+        catalog_version: u64,
+        mutations: impl IntoIterator<Item = Mutation>,
+    ) -> Result<Arc<CatalogState>> {
+        let state = self.mutate(catalog_version, mutations).await?;
+        self.commit_state(catalog_version, state.as_ref().clone())
+            .await
```

Comment:
though it's not _enough_ (I think), if there's going to be a version conflict, we could know/fix it here.

Reply:
i don't understand what you are saying/asking.

Reply:
if `state.version != (catalog_version -1)` then that _might_ (probably?) would be known here and we wouldn't need to run `commit_state()` (because we know it'd error)... similarly, if the mutation applied cleanly, (e.g. 125 does not error), and the commit fails, then it seems like retrying here would be safe.)

Reply:
We already have that check further down `metastore/src/client.rs:332`. 

we _could_ add that check here as well to avoid the retries, but it seems like an optimization.

Reply:
the check in the metastore client (and the retry) doesn't (I  think?) reapply the mutation to the fresh state, so the retry, so while this code, in both cases is _correct_, I'm not sure there's a case where the retry in the metastore client could ever succeed, unless there was some kind of very transient networking problem, and a retry here has some chance of succeeding.

Reply:
removed the retry